paimon-vfs/paimon-vfs-hadoop/src/main/java/org/apache/paimon/vfs/hadoop/PaimonVirtualFileSystem.java

```diff
@@ -0,0 +1,418 @@
+            options.set(RESTCatalogOptions.DLF_ACCESS_KEY_ID, accessKeyId);
+            options.set(RESTCatalogOptions.DLF_ACCESS_KEY_SECRET, accessKeySecret);
+        }
+        String regionId = getConf().get(PaimonVirtualFileSystemConfiguration.FS_DLF_REGION);
+        if (regionId != null) {
+            options.set(RESTCatalogOptions.DLF_REGION, regionId);
+        }
+        // pas://catalog_name/database_name/table_name/file, so uri authority is catalog name
+        options.set(CatalogOptions.WAREHOUSE, uri.getAuthority());
+
+        CatalogContext catalogContext = CatalogContext.create(options);
+        vfsOperations = new VFSOperations(catalogContext);
+    }
+
+    private String getVirtualPath(Path path) {
+        Path qualifiedPath = path.makeQualified(uri, workingDirectory);
+        // convert to absolute path, like /database_name/table_name/file
+        return qualifiedPath.toUri().getPath();
+    }
+
+    @Override
+    public FSDataOutputStream create(
+            Path f,
+            FsPermission permission,
+            boolean overwrite,
+            int bufferSize,
+            short replication,
+            long blockSize,
+            Progressable progress)
+            throws IOException {
+        VFSIdentifier vfsIdentifier = vfsOperations.getVFSIdentifier(getVirtualPath(f));
+        if (vfsIdentifier instanceof VFSCatalogIdentifier) {
+            throw new IOException(
+                    "Cannot create file for virtual path " + f + " which is a catalog");
+        } else if (vfsIdentifier instanceof VFSDatabaseIdentifier) {
+            throw new IOException(
+                    "Cannot create file for virtual path " + f + " which is a database");
+        } else if (vfsIdentifier instanceof VFSTableRootIdentifier) {
+            throw new IOException(
+                    "Cannot create file for table level virtual path " + f + " which is a table");
+        } else {
+            VFSTableObjectIdentifier vfsTableObjectIdentifier =
+                    (VFSTableObjectIdentifier) vfsIdentifier;
+            if (!vfsTableObjectIdentifier.isTableExist()) {
+                throw new IOException(
+                        "Cannot create a file for virtual path "
+                                + f
+                                + " which is not in an existing table");
+            }
+            PositionOutputStream out =
+                    vfsTableObjectIdentifier
+                            .fileIO()
+                            .newOutputStream(vfsTableObjectIdentifier.getRealPath(), overwrite);
+            return new FSDataOutputStream(out, statistics);
+        }
+    }
+
+    @Override
+    public FSDataInputStream open(Path path, int bufferSize) throws IOException {
+        VFSIdentifier vfsIdentifier = vfsOperations.getVFSIdentifier(getVirtualPath(path));
+        if (vfsIdentifier instanceof VFSCatalogIdentifier) {
+            throw new IOException(
```

Comment:
Maybe, here should throw `FileNotFoundException`, can you take a look to the hadoop FileSystem behavior?